File: drouter-plugin-lib/src/main/java/com/didi/drouter/annotation/Service.java
package com.didi.drouter.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by gaowei on 2018/8/30
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.CLASS)
public @interface Service {

    Class<?>[] function();

    String[] alias() default {};

    Class[] feature() default {};

    int priority() default 0;

    int cache() default 0;
}

File: plugins/src/main/kotlin/com/didi/drouter/annotation/Service.kt
package com.didi.drouter.annotation

import kotlin.reflect.KClass

/**
 * Created by gaowei on 2018/8/30
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.BINARY)
annotation class Service(
    val function: Array<KClass<*>>,
    val alias: Array<String> = [],
    val feature: Array<KClass<*>> = [],
    val priority: Int = 0,
    val cache: Int = 0
)
